index.js

```diff
@@ -177,3 +221,45 @@ function findBestSubtitleFile(list) {
     .filter((x) => x.subs)
     .sort((x, y) => x.priority > y.priority)[0];
 }
+
+function processFormats(formats) {
+  const filteredFormats = formats.filter(({ format_id, fps, height, vcodec }) => format_id !== 'source' && !format_id.startsWith('http') && vcodec && vcodec !== 'none' && !vcodec.includes('av01') && !vcodec.includes('vp9') && !vcodec.includes('vp09') && ((height >= 1080 && fps <= 30) || (height < 1080)))
+  .sort((prevFormat, nextFormat) => nextFormat.tbr - prevFormat.tbr);
```

Comment:
This `formats.filter(...).sort(...)` statement is working correctly, since it's returning the right results in all of our test videos. 

But it's a bit confusing exactly what this sort is doing and why you need to do it. 

It may be worth splitting this into two lines of code (1 for the filter, and 1 for the sort) and throwing in a comment.

Reply:
Yep, the existence of this sort is confusing at a glance. I have no idea what it's actually doing or why we even need to do this

Reply:
So certain codecs need to be filtered out since we don't support them and with the height and fps, if the video is 1080p or more then we only want 30fps video but if the resolution is lower than 1080p then we don't mind what the fps is. I then sort by `tbr` which is the bitrate of the format as so that in combination with the `find` method, the highest bitrate format will be picked.

Reply:
You should definitely add a brief comment explaining this

Reply:
And in the comment, might be worth pointing out that in V2, this filtering was done in `service.py` via the arguments passed to yt-dlp. 

In V3, we are instead doing the filtering here, but we are using the same criteria as previous.